### DISCUSSION TITLE
PieCharts Decoupling Data from Color options

### DISCUSSION THREAD

**Discussion by ebratz-aws** [shadcn-ui/ui#4921] [Q&A]
Hi all...

Not sure if possible but I could not find an answer for this:
Is possible to decouple the data from chart configuration in PieCharts?
My goals is to have pure data coming from an API and set the graph properties in another way.

Currently all the Pie Charts have some chartData like:
const chartData = [
  { browser: "chrome", visitors: 275, fill: "var(--color-chrome)" },
  { browser: "safari", visitors: 200, fill: "var(--color-safari)" },
  { browser: "firefox", visitors: 287, fill: "var(--color-firefox)" },
  { browser: "edge", visitors: 173, fill: "var(--color-edge)" },
  { browser: "other", visitors: 190, fill: "var(--color-other)" },
]

Would be great if there's a way to decouple the "fill" property from that dataset. 
Thanks in advance

**Comment by guvenfazli** [comment]
up

**Comment by dzaky-pr** [comment]
i need this answer too

**Comment by Evavic44** [comment]
Recharts provides a `Cell` component that allows you customize the color of each segment in a chart. To apply different colors to each node (e.g., each slice of a pie chart), define an array of colors and map them by index.

```tsx
// Custom array of colors (ensure it covers all data entries)
const COLORS = ["#0088FE", "#00C49F", "#FFBB28", "#FF8042", "#ff4242", "#8a42ff"];
```

A Pie chart example:

```tsx
const chartData = [
  { browser: "chrome", visitors: 275 },
  { browser: "safari", visitors: 200 },
  { browser: "firefox", visitors: 287 },
  { browser: "edge", visitors: 173 },
  { browser: "brave", visitors: 350 },
  { browser: "other", visitors: 190 },
];

const chartConfig = {
  visitors: { label: "Visitors" },
  chrome: { label: "Chrome" },
  safari: { label: "Safari" },
  firefox: { label: "Firefox" },
  edge: { label: "Edge" },
  brave: { label: "Brave" },
  other: { label: "Other" },
} satisfies ChartConfig;

<PieChart>
  <ChartTooltip
    cursor={false}
    content={<ChartTooltipContent hideLabel />}
  />
  <Pie
    data={chartData}
    dataKey="visitors"
    nameKey="browser"
    innerRadius={60}
    strokeWidth={5}
  >
    {chartData.map((entry, index) => (
      <Cell
        key={`cell-${entry.browser}`}
        fill={COLORS[index % COLORS.length]}
      />
    ))}
  </Pie>
</PieChart>
```

<img width="748" height="423" alt="image" src="https://github.com/user-attachments/assets/8bbd67e3-ba32-4642-a7d7-f51bd7144440" />
